docs/conf.py

```diff
@@ -17,20 +18,25 @@
 
 # -- General ------------------------------------------------------------------
 
+sys.path.insert(0, str(Path(__file__).parent / "_extensions"))
+
 extensions = [
     "breathe",
     "myst_parser",
     "sphinx.ext.mathjax",
+    "warnings_filter",
 ]
+
+warnings_filter_config = str(Path(__file__).parent / "known-warnings.txt")
+warnings_filter_silent = True
+
 source_suffix = {
     ".rst": "restructuredtext",
     ".md": "markdown",
 }
 master_doc = "index"
 # ensure the in-source build directory is ignored
-exclude_patterns = [
-    "_build",
-]
+exclude_patterns = ["_build", "api/api_stub.rst", "api/api_index.rst"]
```

Comment:
is the `api/*` stuff used somewhere? if not maybe we can remove it entirely

Reply:
The `index.rst` links out to `api/api.rst`, so that shouldn't be ignored. At the same time, we don't want sphinx to build these extra `.rst` files.